### PR TITLE
Updated plugin by rewriting it in the new seed format

### PR DESCRIPTION
**Pull request by nickdekkers**

### CODE REVIEW

File: src/flashlight.common.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Observable } from 'tns-core-modules/data/observable';
+
+export abstract class Common extends Observable {
```
Comment (dennis-montana) [inline src/flashlight.common.ts]: Why do you extend Observable?

File: src/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        "JavaScript",
+        "Android",
+        "iOS"
+    ],
+    "author": {
+        "name": "Your Name",
+        "email": "youremail@yourdomain.com"
+    },
+    "bugs": {
+        "url": "https://github.com/nickdekkers/nativescript-flashlight/issues"
```
Comment (dennis-montana) [inline src/package.json]: Change these to /Essent instead of /nickdekkers

File: src/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+    "name": "nativescript-flashlight",
```
Comment (dennis-montana) [inline src/package.json]: Why did you remove @essent?

File: src/package.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        "clean": "cd ../demo && rimraf hooks node_modules platforms && cd ../src && rimraf node_modules && npm run plugin.link",
+        "ci.tslint": "npm i && tslint '**/*.ts' --config '../tslint.json' --exclude '**/node_modules/**'"
+    },
+    "keywords": [
+        "NativeScript",
+        "JavaScript",
+        "Android",
+        "iOS"
+    ],
+    "author": {
```
Comment (dennis-montana) [inline src/package.json]: Can you add the old author: "Essent".

File: src/package.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
+{
+    "name": "nativescript-flashlight",
+    "version": "1.0.0",
+    "description": "Your awesome NativeScript plugin.",
```
Comment (dennis-montana) [inline src/package.json]: Add a better description.

File: src/platforms/android/AndroidManifest.xml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+	<!--Describe the permissions, features or other configurations required by your plugin for Android.
```
Comment (dennis-montana) [inline src/platforms/android/AndroidManifest.xml]: Don't we need the camera permissions?

File: LICENSE [hunk 7]
```diff
@@ -1,22 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {yyyy} {name of copyright owner}
```
Comment (dennis-montana) [inline LICENSE]: Are we using the right license?

**Review by dennis-montana — CHANGES_REQUESTED** [review]

File: LICENSE (hunk 7 above)
Comment (corne-de-bruin) [reply LICENSE]: Just add the old version again

File: demo/app/main-page.ts [hunk 8]
```diff
@@ -1,32 +1,32 @@
-import { FlashLight } from '@essent/nativescript-flashlight';
-import { EventData, Observable, PropertyChangeData } from 'data/observable';
-import { Page } from 'ui/page';
-import { Slider } from 'ui/slider';
+import { Flashlight } from 'nativescript-flashlight';
```
Comment (corne-de-bruin) [inline demo/app/main-page.ts]: This should be @essent/nativescript-flashlight

File: src/flashlight.common.ts (hunk 1 above)
Comment (nickdekkers) [reply src/flashlight.common.ts]: I did not personally do this. This is how the new seed did this. So instead of touching it, I thought of letting it stay there

File: demo/app/main-page.ts (hunk 8 above)
Comment (nickdekkers) [reply demo/app/main-page.ts]: It will not recognize the module if I add @essent in front of it

**Review by corne-de-bruin — APPROVED** [review]

**Review by dennis-montana — APPROVED** [review]
